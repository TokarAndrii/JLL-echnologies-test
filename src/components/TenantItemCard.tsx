import { makeStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import Typography from "@material-ui/core/Typography";
import Divider from "@material-ui/core/Divider";

export type TenantItemData = {
  id: string;
  name: string;
  description: string;
  code: string;
  type: string;
  status: string;
};

const useStyles = makeStyles({
  root: {
    maxWidth: "30vw",
    margin: "20px",
    padding: "15px",
    cursor: "pointer"
  },
  bullet: {
    display: "inline-block",
    margin: "0 2px",
    transform: "scale(0.8)"
  },
  title: {
    fontSize: 16,
    fontWeight: "bold"
  },
  info: {
    padding: 5
  },
  description: {
    padding: 5,
    marginTop: 15
  }
});

const TenantItemCard = ({
  data,
  onClick
}: {
  data: TenantItemData;
  onClick: () => void;
}) => {
  const classes = useStyles();
  const bull = <span className={classes.bullet}>â€¢</span>;

  const { name, description, code, type, status } = data;

  return (
    <Card className={classes.root} onClick={onClick}>
      <Typography className={classes.title} color="textSecondary" gutterBottom>
        {name}
      </Typography>
      <Divider />
      <Typography component="h2" className={classes.info}>
        {bull}code: {code}; {bull}type: {type}; {bull}status: {status}
      </Typography>
      <Divider />
      <Typography component="p" className={classes.description}>
        {description}
      </Typography>
    </Card>
  );
};

export default TenantItemCard;
