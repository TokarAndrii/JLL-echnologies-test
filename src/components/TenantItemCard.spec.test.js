import { shallow } from "enzyme";
import TenantItemCard from "./TenantItemCard";

const TENANT_ITEM_DATA = {
  id: "id",
  name: "name",
  description: "description",
  code: "code",
  type: "type",
  status: "status"
};

const TEST_HANDLE_CLICK = () => console.log("CLICK");

describe("TenantItemCard component", () => {
  it("should render TenantItemCard", () => {
    const component = shallow(
      <TenantItemCard data={TENANT_ITEM_DATA} onClick={TEST_HANDLE_CLICK} />
    );
    expect(component).toMatchSnapshot();
  });
  it("should render TenantItemCard once", () => {
    const component = shallow(
      <TenantItemCard data={TENANT_ITEM_DATA} onClick={TEST_HANDLE_CLICK} />
    );
    const wrapper = component.find(".makeStyles-root-1");
    expect(wrapper.length).toBe(1);
  });
});
