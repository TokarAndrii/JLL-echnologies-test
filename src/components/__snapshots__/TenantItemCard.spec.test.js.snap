// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TenantItemCard component should render TenantItemCard 1`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-root-1"
  onClick={[Function]}
>
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-title-3"
    color="textSecondary"
    gutterBottom={true}
  >
    name
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Divider)) />
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-info-4"
    component="h2"
  >
    <span
      className="makeStyles-bullet-2"
    >
      •
    </span>
    code: 
    code
    ; 
    <span
      className="makeStyles-bullet-2"
    >
      •
    </span>
    type: 
    type
    ; 
    <span
      className="makeStyles-bullet-2"
    >
      •
    </span>
    status: 
    status
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Divider)) />
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-description-5"
    component="p"
  >
    description
  </WithStyles(ForwardRef(Typography))>
</WithStyles(ForwardRef(Card))>
`;
